### ISSUE TITLE
How to draw the loss landscape for sampled models

### ISSUE THREAD

**Issue by aminullah6264** [wpeebles/G.pt#4]
Can you please guide me on how to draw Figure 8 in the original paper? : G.pt learns a multimodal distribution over local error minima.

**Comment by wpeebles** [comment]
Hi @Aminullah6264, sorry for the delay in responding! Here's the rough outline of how we generated each subplot in Figure 8:

1. Randomly-sample one (unseen) starting parameter vector from the test split of our dataset. I.e., this a randomly-initialized parameter vector.
2. Use the pre-trained G.pt model to sample 256 different updated parameter vectors (all using the same starting parameters and varying sampling noise).
3. Run PCA on the batch of sampled parameter vectors (reducing to two dimensions).
4. Generate a loss/error/return/etc. 3D surface plot centered at the mean parameter vector, traversing in the two PCA directions (we used [plotly's `go.Surface`](https://plotly.com/python/3d-surface-plots/) for the actual plotting).
5. Project all of G.pt's parameter samples into two-dimensional PCA space.
6. Remove any of the 256 parameter samples that were poorly-modeled by PCA (more than 1e-3 L2 reconstruction loss in parameter space).
7. Compute the test loss/error/return/etc. of G.pt's *PCA-projected* parameter samples.
8. Overlay the samples' losses/errors/returns as points on top of the surface plot.

Hope this helps!
